src/Backoffice/Task/Domain/Actions/UpsertTaskAction.php

```diff
@@ -0,0 +1,32 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Lightit\Backoffice\Task\Domain\Actions;
+
+use Lightit\Backoffice\Task\Domain\DataTransferObjects\TaskDto;
+use Lightit\Backoffice\Task\Domain\Models\Task;
+
+class UpsertTaskAction
+{
+    public function execute(TaskDto $taskDto): Task
+    {
+        if ($taskDto->taskId) {
+            $task = Task::query()->find($taskDto->taskId);
+            if (! $task) {
+                throw new \Exception('Task not found');
+            }
```

Comment:
```suggestion
            $task = Task::query()->findOrFail($taskDto->taskId);
```

Shouldn't be necessary to validate if it exists since you did it on the Form Request